### ISSUE TITLE
Use two separate models : one for WM & GM and one for the lesions

### ISSUE THREAD

**Issue by Luugaaa** [ivadomed/model_seg_sc-gm-lesion_human_ms_exvivo_t2star#28]
# Context 

We investigated the use of two separate models to segment the spinal cord: one to segment the white matter and the gray matter and one to segment only the lesions (the lesions are separated into two categories, lesions on white matter and lesions on gray matter). 

# Method 

We used the best base configuration known to this point: mag and phase in two channels and otsu thresholding for preprocessing #24 . 
Since we got worse results when training models to segment only lesions, we didn't run the experiment using only WM and GM ground truth.

# Results

Note: the results presented below are dice scores obtained on a 12 subjects dataset on the validation set and are the average on 5 folds.

The colors are: 
- **orange:** models trained on lesions only
- **red:** base models (trained on WM, GM and lesions)

## Class 3: lesions on WM

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/e6890230-9e99-46df-bc82-6162d4aa3724" />

** Observation:** The model trained on lesions only is slightly underperforming 

## Class 4: lesions on GM

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/122039fa-8c72-4c1d-b016-c5aab0a9e0a5" />

**Observation:** The model trained on lesions only is underperforming significantly (around -10%).

# Conclusion

We will keep working with a model trained to segment WM, GM and lesions simultaneously. I believe the reason why the lesion specialized model underperformed is that the more a model knows about the objects in the image it's segmenting, the better it will perform. For example, when segmenting gray matter with some hyperintensity, the specialized model has to decide between lesion and background, while the more "generalist" model may have already learned that in some contexts a part of the gray matter can be hyperintense, because it's also trained (--training) to segment gray matter.
